### ISSUE TITLE
[$500] [Wave 8] [Ideal Nav] Android - "Welcome to Expensify" pop up does not appear when navigate to Concierge link

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#35663]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.35-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4265030&group_by=cases:section_id&group_id=229066&group_order=asc
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Log out of NewDot
2. Navigate to https://staging.new.expensify.com/concierge
3. Sign-up with a new account

## Expected Result:
"Welcome to Expensify" pop up should appear

## Actual Result:
"Welcome to Expensify" pop up does not appear when new user navigate to New Dot via Concierge link

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/888d52e9-dcfb-4b90-8882-3bb2b5ad193f

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a54484263942f220</li>
        <li>Upwork Job ID: 1753414483477962752</li>
        <li>Last Price Increase: 2024-02-02</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Reviewer | 28144631</li>
<li>DylanDylann | Contributor | 28144633</li>
</ul></li>
    </ul>
</details>

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- Android - "Welcome to Expensify" pop up does not appear when navigate to Concierge link
### What is the root cause of that problem?
- We have a logic that check should we open the Welcome modal or not:
https://github.com/Expensify/App/blob/3ab4e6e12d9ff38e69493321f0d31f032481b373/src/components/PurposeForUsingExpensifyModal.tsx#L102-L104
- In this case, ```NAVIGATORS.BOTTOM_TAB_NAVIGATOR``` is not equal to ```currentRoute.name``` that leads to the bug,
### What changes do you think we should make in order to solve the problem?
- We can create the list routes that we want to display the Welcome modal and use in the above condition

### What alternative solutions did you explore? (Optional)
- NA

**Comment by shubham1206agra** [comment]
@DylanDylann Can you create a test branch for the same?

**Comment by jeremy-croff** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The welcome modal does not show anymore in browser views.

### What is the root cause of that problem?
https://github.com/Expensify/App/pull/33280/commits/5899cfff6a43698bd424cee9b5691d1749aa7456 changed the route that triggers the welcome modal to be BOTTOM_TAB_NAVIGATOR. Which only works for mobile views.


### What changes do you think we should make in order to solve the problem?
Additionally to checking if the route is BOTTOM_TAB_NAVIGATOR,  we should show the modal if it matches the previous condition, CENTRAL_PANE_NAVIGATOR. Which will work for web views.

### What alternative solutions did you explore? (Optional)

**Comment by shubham1206agra** [comment]
@hayata-suenaga This might be related to Ideal Nav by seeing the solution now. This bug existed earlier too, but now the solution might be closely related to Ideal Nav as whole navigation changed.

**Comment by DylanDylann** [comment]
@shubham1206agra I do not available to create test branch now. Generally, we can update the condition to:
```
if (currentRoute && NAVIGATORS.BOTTOM_TAB_NAVIGATOR !== currentRoute.name && "Concierge" !== currentRoute.name) { 
     return; 
 } 
```

**Comment by hayata-suenaga** [comment]
> This bug existed earlier too, but now the solution might be closely related to Ideal Nav as whole navigation changed. Can you put this in Wave 8 in this case?

Yes this seems reasonable. I'll add this to ideal nav. Thank you for bringing this to my attention 🙇

**Comment by hayata-suenaga** [comment]
I'll keep this issue as external, but could you, @adamgrzybowski, review the proposals as you know the new navigation system the best? Or if you can raise PR, that would be helpful, too 🙇

**Comment by mountiny** [comment]
@shubham1206agra Can you also please review the proposal from @bernhardoj on this dupe issue https://github.com/Expensify/App/issues/35748#issuecomment-1925641555

And review the proposals made here? while we can say its not a blocker it would be great to fix this asap

**Comment by shubham1206agra** [comment]
@mountiny I am just waiting @adamgrzybowski to respond here as per https://github.com/Expensify/App/issues/35663#issuecomment-1925586968

**Comment by bernhardoj** [comment]
https://github.com/Expensify/App/issues/35748 is actually not a dupe of this one. This issue exists before the ideal nav, but https://github.com/Expensify/App/issues/35748 happens after the ideal nav.

Previously, we showed the engagement model only if the active route is either a HOME or CENTRAL_PANE_NAVIGATOR. If we visit /concierge, the active route will be ConciergePage and it's never included in the engagement/welcome logic which is what this issue is about. 

But after the ideal nav, we replace both HOME and CENTRAL_PANE_NAVIGATOR with BOTTOM_TAB_NAVIGATOR. BOTTOM_TAB_NAVIGATOR is the replacement for HOME, but since CENTRAL_PANE_NAVIGATOR is removed, the engagement modal never shows if we normally sign up on the web and that's what https://github.com/Expensify/App/issues/35748 is about. My solution in https://github.com/Expensify/App/issues/35748 won't solve this issue.

**Comment by hayata-suenaga** [comment]
ah I see. I agree with you, @bernhardoj. If @shubham1206agra also confirms this, we can reopen https://github.com/Expensify/App/issues/35748 and @bernhardoj can open a PR over there.

**Comment by shubham1206agra** [comment]
@bernhardoj Your solution is very similar to https://github.com/Expensify/App/issues/35663#issuecomment-1924999723 here, and I think it will be duplicated efforts for small gains. If @hayata-suenaga agrees, we can do both fixes here. If not, you can reopen the issue.

**Comment by DylanDylann** [comment]
@mountiny Pls note that my proposal [here](https://github.com/Expensify/App/issues/35663#issuecomment-1924999723) come first and also fix the issue so I can create a quick PR for this

**Comment by jeremy-croff** [comment]
I do not see this as separate issues, I have tested and confirmed that the concierge route will go through CENTRAL_PANE_NAVIGATOR navigator, after the Concierge Route.

Even Running the fix locally will show the welcome modal after including CENTRAL_PANE_NAVIGATOR, and signing up thru https://staging.new.expensify.com/concierge.  

@bernhardoj proposals and mine  were the same, after identifying the same root cause. However, my stance it that it does fix both tickets after testing and confirming.

@DylanDylann fix does not identify the root cause which is why it would only solve this ticket.

**Comment by DylanDylann** [comment]
@jeremy-croff When proposing the solution, I just aim to fix the particular bug of this issue, that is with ```/Concierge``` page. And I`ve mentioned in my proposal, "We can create the list routes that we want to display the Welcome modal and use in the above condition", Because there are a lot of routes that we want to display the welcome modal, not only ```/Concierge```, for example, 
```/r/:id```, ```/a/:id```, ...

**Comment by shubham1206agra** [comment]
@mountiny Feel free to assign anyone here.

**Comment by hayata-suenaga** [comment]
I agree! let's handle both issues in this issue.

I'm assigning @DylanDylann here as their proposal came first and they did mention that we can list routes where we want to display the Welcome flow.

However, thank you @jeremy-croff for your proposal 🙇 I hope we can work together next time 😄

**Comment by hayata-suenaga** [comment]
assigning myself

**Comment by jeremy-croff** [comment]
Just be noted that 
CENTRAL_PANE_NAVIGATOR and BOTTOM_TAB_NAVIGATOR solves for concierge route in mobile and web Adding Concierge route will be a redudancy in that case, as the concierge route is on CENTRAL_PANE_NAVIGATOR. 

BOTTOM_TAB_NAVIGATOR + Concierge = insufficient
BOTTOM_TAB_NAVIGATOR + CENTRAL_PANE_NAVIGATOR + Concierge is redudant 
BOTTOM_TAB_NAVIGATOR + CENTRAL_PANE_NAVIGATOR handles the new nav flow in it's entirety

**Comment by DylanDylann** [comment]
@jeremy-croff Thanks for your [suggestion](https://github.com/Expensify/App/issues/35663#issuecomment-1928499794)

**Comment by marcaaron** [comment]
This still works correctly on production so I'm adding a Deploy Blocker label. 

@hayata-suenaga Can you give an ETA on this one?

**Comment by jeremy-croff** [comment]
> ~This still works correctly on production so I'm adding a Deploy Blocker label.~
> 
> ~@hayata-suenaga Can you give an ETA on this one?~
> 
> Oh hmm actually it looks like maybe the behavior is inconsistent after all :/ Just re-tested and the modal came up for both staging and production 😅

I can confirm that on mobile or small screen it would show up, larger screen it wont.

**Comment by marcaaron** [comment]
@hayata-suenaga What is the status of this issue and ETA for a fix?

**Comment by marcaaron** [comment]
Hi, Since these changes are required to unblock our PR [here](https://github.com/Expensify/App/pull/35019) I proposed that we move this change directly into that PR. We will still compensate @DylanDylann for the solution. But figured we'd save some time by addressing it in the other PR. Thanks for the help on this one! (I will close the other PR now)

**Comment by jeremy-croff** [comment]
> Hi, Since these changes are required to unblock our PR [here](https://github.com/Expensify/App/pull/35019) I proposed that we move this change directly into that PR. We will still compensate @DylanDylann for the solution. But figured we'd save some time by addressing it in the other PR. Thanks for the help on this one! (I will close the other PR now)

@marcaaron @hayata-suenaga 
I humbly request to review my proposal https://github.com/Expensify/App/issues/35663#issuecomment-1925066729
Which correctly identified the root cause, as per a regression. As well as proposed the actual, exact solution.  which is what ended up being committed by @teneeto https://github.com/Expensify/App/pull/35019/commits/dd69810b7acf939559beb439eb990ad2a25516ac to unblock the deployment.

You can compare it against the current assigned proposal https://github.com/Expensify/App/issues/35663#issuecomment-1924999723 which did not identify the root cause, nor the solution. Since he now also isn't the contributor anymore either, the compensation doesn't seem to be merited on all accounts.

**Comment by marcaaron** [comment]
Of course, the issue just closes when we merged the other PR automatically. I reopened it. @hayata-suenaga or @conorpendergrast can you respond to these concerns? Thanks!

**Comment by DylanDylann** [comment]
> since he now also isn't the contributor anymore either, the compensation doesn't seem to be merited on all accounts
 
- FYI, The contributor [guideline](https://github.com/DylanDylann/App/blob/main/contributingGuides/HOW_TO_BECOME_A_CONTRIBUTOR_PLUS.md) mentioned that C+ can also work on jobs as a contributor

> the current assigned proposal https://github.com/Expensify/App/issues/35663#issuecomment-1924999723 did not identify the root cause, nor the solution

- This issue is just related to the "Concierge link" (based on the title), so my solution just aims to fix that. I also suggested in the [proposal](https://github.com/Expensify/App/issues/35663#issuecomment-1924999723) that, we should add the list routes we want to display the "Welcome to Expensify" modal, which is very straightforward. It is also mentioned in [internal `s comment](https://github.com/Expensify/App/issues/35663#issuecomment-1928483787)

**Comment by hayata-suenaga** [comment]
@shubham1206agra can I confirm that @DylanDylann's proposal was used in the other PR?
also, @shubham1206agra, did you review the other PR?

**Comment by hayata-suenaga** [comment]
let's do partial payments for both @shubham1206agra as C+ as they reviewed proposals and @DylanDylann for their proposal

**Comment by DylanDylann** [comment]
@laurenreidexpensify The PR is closed, I think we should add Daily label to move forward this one

**Comment by hayata-suenaga** [comment]
@laurenreidexpensify please see [this message](https://github.com/Expensify/App/issues/35663#issuecomment-1944335693) for payment

**Comment by laurenreidexpensify** [comment]
Payment Summary:

Contributor: $250 - @DylanDylann paid in Upwork 
C+: $250 - @shubham1206agra offer sent in Upwork, waiting to accept

**Comment by laurenreidexpensify** [comment]
Final Payment Summary:

Contributor: $250 - @DylanDylann paid in Upwork
C+: $250 - @shubham1206agra paid in Upwork
